### DISCUSSION TITLE
Generating Dynamic URL Parameters

### DISCUSSION THREAD

**Discussion by ghost** [47ng/nuqs#607] [Q&A] [answered]
How can I generate dynamic URL parameters based on different product options like size, color, and material? For example, if I have a shirt with options like size and color, the URL should look like this: `?size=M&color=black`.

**Comment by franky47** [comment, marked as answer]
The [serializer helper](https://nuqs.47ng.com/docs/utilities#serializer-helper) is a good place to start.

Iterate over your options, pass it to the serialize method it returns, and you'll get the corresponding URL query string (with an optional base path).
